test/controllers/knowledgeBase.controller.test.js

```diff
@@ -496,14 +469,16 @@ describe('Get one post', () => {
   context('with a valid token & id', () => {
     it('returns successful payload', (done) => {
       request()
-        .get(`/api/v1/knowledge-base/${postId}`)
-        .set('authorization', adminToken)
+        .get(`/api/v1/knowledge-base/${slug}`)
         .end((err, res) => {
           expect(res).to.have.status(200);
           expect(res.body.success).to.be.eql(true);
           expect(res.body).to.have.property('post');
           expect(res.body.post).to.have.property('title');
           expect(res.body.post).to.have.property('author');
+          expect(res.body.post.author).to.have.property('firstName');
```

Comment:
You need to generate an admin from the user factory, use the admin id to create a post and verify that the firstName returned here actually corresponds with the generated one.

Here is a better way to ensure that the right value is returned.

```suggestion
          expect(res.body.post.author.firstName).to.eql(admin.firstName);
```